File: Codigos estudiantes por lenguaje/Java/Daniel Cangrejo/Algoritmos_de_Busqueda/minAbsSumPair.java

public class Busquedas6 {
	
	static void minAbsSumPair(int arr[], int n) {
		
		int left, right, min_sum, sum, minl, minr;
		if(n < 2) {
			System.out.println("Invalido");
			return;
		}
		minl = 0;
		minr = 1;
		min_sum = arr[0] + arr[1];
		for (left = 0; left < n-1; left++) {
			for (right = left+1; right < n; right++) {
				sum = arr[left] + arr[right];
				if(Math.abs(min_sum) > Math.abs(sum)) {
					min_sum = sum;
					minl = left;
					minr = right;
				}
			}
		}
		System.out.println(" Los dos elementos los cuales "
				+ "tienen la suma minima son "
				+ arr[minl] + " y " + arr[minr] + " = " + min_sum);
	}

	public static void main(String[] args) {
		int arr[] = {1, 60, -10, 70, -80, 85};
		minAbsSumPair(arr, 6);

	}

}

File: Codigos estudiantes por lenguaje/C++/Daniel Cangrejo/Búsquedas/minAbsSumPair.cpp
#include <bits/stdc++.h>
#include <cstdlib>
using namespace std;
void minAbsSumPair(int arr[],int n){
	int l,r,min_sum,sum,min_l,min_r;
	if(n<2){
		printf("Entrada no valida");
	}
	min_l=0;
	min_r=1;
	min_sum=arr[0]+arr[1];
	for(l=0;l<n-1;++l){
		for(r=l+1;r<n;++r){
			sum=arr[l]+arr[r];
			if(std::abs(min_sum)>std::abs(sum)){
				min_sum=sum;
				min_l=l;
				min_r=r;
			}
		}
	}
	printf("Los numero que tienen la minima suma son %d %d",arr[min_l],arr[min_r]);
}
int main(int argc, char const *argv[]) {
	int arr[]={1,-2,-10,70,-80,85};
	int n =sizeof(arr)/sizeof(arr[0]);
	minAbsSumPair(arr,n);

    return 0;
}
